core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/MultiregionalBucketMigrationActor.scala

```diff
@@ -0,0 +1,857 @@
+        )
+      )
+    } yield ()
+
+  // Create bucket in us-central1-a
+  final def createBucketInTargetRegion(migration: MultiregionalBucketMigration,
+                                       workspace: Workspace,
+                                       sourceBucketName: GcsBucketName,
+                                       destBucketName: GcsBucketName
+  ): MigrateAction[Unit] =
+    MigrateAction { env =>
+      def pollForBucketToBeCreated(interval: FiniteDuration, deadline: Deadline): IO[Unit] =
+        IO.sleep(interval).whileM_ {
+          for {
+            _ <- IO.raiseWhen(deadline.isOverdue()) {
+              WorkspaceMigrationException(
+                message = "Workspace migration failed: timed out waiting for bucket creation",
+                data = Map(
+                  "migrationId" -> migration.id,
+                  "workspace" -> workspace.toWorkspaceName,
+                  "googleProject" -> workspace.googleProjectId,
+                  "bucketName" -> destBucketName
+                )
+              )
+            }
+
+            // Don't need a requester pays project for the bucket in the new google project
+            // as requester pays if enabled at the end of the migration, if at all.
+            bucket <- env.storageService.getBucket(
+              GoogleProject(workspace.googleProjectId.value),
+              destBucketName
+            )
+          } yield bucket.isEmpty
+        }
+
+      OptionT.liftF {
+        for {
+          sourceBucketOpt <- env.storageService.getBucket(
+            GoogleProject(workspace.googleProjectId.value),
+            sourceBucketName,
+            bucketGetOptions =
+              if (migration.requesterPaysEnabled) List(BucketGetOption.userProject(env.googleProjectToBill.value))
+              else List.empty
+          )
+
+          sourceBucket = sourceBucketOpt.getOrElse(
+            throw WorkspaceMigrationException(
+              message = "Workspace migration failed: source bucket not found.",
+              data = Map(
+                "migrationId" -> migration.id,
+                "workspace" -> workspace.toWorkspaceName,
+                "sourceBucket" -> sourceBucketName
+              )
+            )
+          )
+
+          _ <- env.storageService
+            .insertBucket(
+              googleProject = GoogleProject(workspace.googleProjectId.value),
+              bucketName = destBucketName,
+              labels = Option(sourceBucket.getLabels).map(_.asScala.toMap).getOrElse(Map.empty),
+              bucketPolicyOnlyEnabled = true,
+              logBucket = GcsBucketName(
+                GoogleServicesDAO.getStorageLogsBucketName(workspace.googleProjectId)
+              ).some,
+              location = env.defaultBucketLocation.some,
+              autoclassEnabled = true
+            )
+            .compile
+            .drain
+
+          _ <- pollForBucketToBeCreated(interval = 100.milliseconds, deadline = 10.seconds.fromNow)
```

Comment:
Was the 10.seconds value take straight from PPW? I'm wondering if that needs to be more generous.

Reply:
Yeah, this was taken directly. I'm not sure if PPW ran into any issues with this timing out, but I see no reason not to bump it up to 30 seconds to give us some extra breathing room